### PR TITLE
gr-filter: C++ generation

### PR DESCRIPTION
**Pull request by seeAppRun**
Fixed issues in some gr-filter yml files.
Added c++ support to some filter blocks,
Re-factored taps generation in pfb.py to allow c++ code to leverage default taps values. 

Note these changes are dependent on some fixes in pull request #2797 'grc/cppgen: C++ code generation fixes'

### CODE REVIEW

**Comment by haakov** [comment]
Awesome work, I'll test this tomorrow. I'm not too familiar with `pfb.py`, but from what I can tell you've mostly just moved code around in that file, so that's okay I guess.

**Comment by seeAppRun** [comment]
> Awesome work, I'll test this tomorrow. I'm not too familiar with `pfb.py`, but from what I can tell you've mostly just moved code around in that file, so that's okay I guess.

Yes. I refactored the code to make the taps generation methods static so they could be invoked without instantiating the filter class. This allows the taps values to be used for C++ flowgraphs as well (from the corresponding YAML files).

I know this may not be the most elegant solution but it utilizes the same logic for both Python and C++ flowgraphs. And unfortunately, some of the filter blocks are of no use without valid Taps values.

I wonder if the taps generation could eventually be integrated into the gr-filter library instead of in the Python module? But I guess that's a whole other discussion.

File: gr-filter/grc/filter_low_pass_filter.block.yml [hunk 1]
```diff
@@ -76,21 +76,21 @@ templates:
         ${beta}))
 
 cpp_templates:
-    includes: ['#include <gnuradio/filter/firdes.h>']
-    declarations: 'filter::firdes::sptr ${id};'
+    includes: ['#include <gnuradio/filter/firdes.h>', '#include <gnuradio/filter/interp_fir_filter.h>']
```
[the remark this answers is not in the document]
Comment (seeAppRun) [reply gr-filter/grc/filter_low_pass_filter.block.yml]: > <gnuradio/filter/fir_filter_blk.h>

Good catch!

File: gr-filter/grc/filter_pfb_synthesizer.block.yml [hunk 2]
```diff
@@ -60,16 +60,17 @@ templates:
 
 cpp_templates:
     includes: ['#include <gnuradio/filter/pfb_synthesizer_ccf.h>']
-    declarations: 'pfb::synthesizer_ccf::sptr ${id};'
+    declarations: 'gr::filter::pfb_synthesizer_ccf::sptr ${id};'
     make: |-
-        std::vector<float> taps = {${str(taps)[1:-1]}};
+        <% from gnuradio import filter %>
```
[the remark this answers is not in the document]
Comment (seeAppRun) [reply gr-filter/grc/filter_pfb_synthesizer.block.yml]: > Does this do anything?
No it does not. Good find. The import is not needed here and should be removed.

**Comment by seeAppRun** [comment]
> I left a few comments. Otherwise, great stuff!

@haakov I've made a few changes based on your comments. Should I push these now or wait until the other reviewers have finished?

**Comment by haakov** [comment]
You can push them :)

**Comment by mbr0wn** [comment]
Looks like merging #2797 causes this to conflict. Can you please rebase?

**Comment by mbr0wn** [comment]
Rebased and moved to here: https://github.com/gnuradio/gnuradio/pull/3350

**Pull request closed without merging**